conda/core/index.py

```diff
@@ -51,7 +58,457 @@ def check_allowlist(channel_urls: list[str]) -> None:
+        if prefix_prec:
+            if prec:
+                if prec.channel == prefix_prec.channel:
+                    link = prefix_prec.get("link") or EMPTY_LINK
+                    prec = PrefixRecord.from_objects(prec, prefix_prec, link=link)
+                else:
+                    prefix_channel = prefix_prec.channel
+                    prefix_channel._Channel__canonical_name = prefix_channel.url()
+                    del prefix_prec._PackageRecord__pkey
+                    prec = prefix_prec
+            else:
+                prec = prefix_prec
+        return prec
+
+    def _update_from_cache(self, key, prec):
+        for pcrec in PackageCacheData.get_all_extracted_entries():
+            if pcrec == key:
+                if prec:
+                    # The downloaded repodata takes priority
+                    return PackageCacheRecord.from_objects(prec, pcrec)
+                else:
+                    return pcrec
+        return prec
+
+    def __getitem__(self, key):
+        assert isinstance(key, PackageRecord)
+        try:
+            return self._data[key]
+        except AttributeError:
+            pass
+        try:
+            return self.system_packages[key]
+        except KeyError:
+            pass
+        if self.track_features and key.name.endswith("@"):
+            for feature in self.track_features:
+                if feature == key.name[:-1]:
+                    return make_feature_record(feature)
+        prec = self._retrieve_from_channels(key)
+        prec = self._update_from_prefix(key, prec)
+        if self.use_cache:
+            prec = self._update_from_cache(key, prec)
+        if prec is None:
+            raise KeyError((key,))
+        return prec
+
+    def __contains__(self, key):
+        try:
+            _ = self[key]
+            return True
+        except (PackagesNotFoundError, KeyError):
+            return False
+
+    def __copy__(self):
+        inst = self.__class__.__new__(self.__class__)
+        inst.__dict__.update(self.__dict__)
+        if "_data" in self.__dict__:
+            inst.__dict__["_data"] = self.__dict__["_data"].copy()
+        return inst
+
+
+class ReducedIndex(Index):
+    def __init__(
+        self,
+        specs,
+        channels=(),
+        prepend=True,
+        platform=None,
+        subdirs=None,
+        use_local=False,
+        use_cache=None,
+        prefix=None,
+        repodata_fn=context.repodata_fns[-1],
+        add_system=False,
+    ) -> None:
+        super().__init__(
+            channels,
+            prepend,
+            platform,
+            subdirs,
+            use_local,
+            use_cache,
+            prefix,
+            repodata_fn,
+            add_system,
+        )
+        self.specs = specs
+        self._derive_reduced_index()
+
+    def __repr__(self):
+        channels = ", ".join(self.channels.keys())
+        return f"ReducedIndex(spec={self.specs}, channels=[{channels}])"
```

Comment:
```suggestion
        return f"<ReducedIndex(spec={self.specs}, channels=[{channels}])>"
```

Reply:
As as above, let's use the concrete class name here.